SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/TestActors/TestMovementCharacter.cpp

```diff
@@ -41,6 +44,55 @@ ATestMovementCharacter::ATestMovementCharacter()
 	CameraComponent->SetRelativeRotation(CameraRotation);
 #endif
 	CameraComponent->SetupAttachment(GetCapsuleComponent());
+
+	UCharacterMovementComponent* CharacterMovementComponent = GetCharacterMovement();
+	CharacterMovementComponent->MaxSimulationIterations = 14;
+
+	SpeedWindow.SetNum(SpeedWindowSize);
+}
+
+void ATestMovementCharacter::BeginPlay()
+{
+	Super::BeginPlay();
+	PreviousLocation = GetActorLocation();
+}
+
+void ATestMovementCharacter::Tick(float DeltaTime)
+{
+	float Speed = FVector(GetActorLocation() - PreviousLocation).Size() / DeltaTime;
+
+	if (SpeedWindowIndex > SpeedWindowSize - 1)
```

Comment:
```suggestion
	if (SpeedWindowIndex >= SpeedWindowSize )
```

Reply:
I would normally do `SpeedWindowIndex = (SpeedWindowIndex + 1) % SpeedWindowSize;`, but I know some people have expressed unreserved disgust at such a method, so leaving it here as an option.